clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/LeaderFollowerStoreIngestionTask.java

```diff
@@ -285,6 +287,9 @@ public LeaderFollowerStoreIngestionTask(
             .setPartitionCount(storeVersionPartitionCount)
             .build();
     this.veniceWriter = Lazy.of(() -> veniceWriterFactory.createVeniceWriter(writerOptions));
+    this.maxRecordSizeBytes = (store.getMaxRecordSizeBytes() < 0) // TODO: move to VeniceWriter when nearline supported
```

Comment:
I assume the store-level/default config are only for chunk enabled stores, right?

Reply:
The configs will always be set on the object, but it should only take effect for chunk-enabled stores: ([VeniceWriter](https://github.com/linkedin/venice/blob/4a20cdf03d362989bfcdf0ee22ffd7f2038fe50a/internal/venice-common/src/main/java/com/linkedin/venice/writer/VeniceWriter.java#L911))

Reply:
But in `LFSIT`, we are doing the size check/ratio calculation before passing it to `VeniceWriter`, right?

Reply:
It is true that in `LFSIT`, we are doing the size check/ratio calculation on the Kafka message before passing the message to `VeniceWriter`.

Regarding configs, the store-level/default config will be retrieved from `VeniceServerConfig` by `LFSIT` and passed to `VeniceWriter` via `VeniceWriterOptions` in the constructor of `LFSIT`.

Reply:
> if the store doesn't have chunked enabled, this.maxRecordSizeBytes will be using the default size: 100MB even for non-chunking stores, is it right?

This is correct. 

> Maybe we shouldn't need to emit the size ratio metrics for non-chunking stores?

The metric should already not be emitted for non-chunking stores because it's dependent on detecting `ChunkedValueManifest`, which should not exist for non-chunking stores:

```java
if (put.getSchemaId() == AvroProtocolDefinition.CHUNKED_VALUE_MANIFEST.getCurrentProtocolVersion()) {
  try {
    ChunkedValueManifest chunkedValueManifest = manifestSerializer.deserialize(put.getPutValue());
    if (isValueManifest) {
      recordAssembledRecordSizeRatio();
      hostLevelIngestionStats.recordAssembledValueSize();
    } else { // the manifest pertains to a chunked rmd
      hostLevelIngestionStats.recordAssembledRmdSize();
    }
  }
}
```

Reply:
Got it, thanks.